packages/engine/lib/error/src/report.rs

```diff
@@ -15,34 +15,31 @@ use crate::{
 };
 use crate::{iter::Frames, Context};
 
-/// Contains a [`Frame`] stack consisting of an original error, context information, and optionally
+/// Contains a [`Frame`] stack consisting of [`Context`]s, attachments, and optionally
 /// a [`Backtrace`] and a [`SpanTrace`].
```

Comment:
```suggestion
/// Contains a [`Frame`] stack consisting of [`Context`]s, attachments which optionally
/// include a [`Backtrace`] and a [`SpanTrace`].
```
? As in are we storing the `Backtrace` and `SpanTrace` as attachments (things inside providers)? If we're not then ignore suggestion but I wouldn't think they'd be on the stack.

Reply:
Depends 😄
If the first context provides them,
They will be used from the context,
Otherwise, they will be captured by

Reply:
I put that in the documentation: 028ccfe72fcf2837343d64711514c5e15c77b9dc and b2bc61ed60de3f764e03687f9417bc3ee3007d30

Reply:
I don't think you have put this in the docs.

> are we storing the Backtrace and SpanTrace as attachments (things inside providers)

I don't see an answer of that in the comment. I also don't understand the main phrasing here.

> /// If the root [`Frame`] contains a [`Backtrace`]/[`SpanTrace`], these are used, otherwise they
/// are eventually captured.

Can we please be a bit more specific. 

If the root Frame contains a Backtrace/SpanTrace then what? (What does "these are used" mean?)

What does "they are eventually captured" mean?

Immediately after that you say that you have to **enable** capturing them, so they're not always eventually captured?

Reply:
It's not guaranteed, that a backtrace/spantrace can be captured (even if the features are set), this depends on the platform. I rephrased the whole section to make this clear: 069bb525a5cdc435827732ae3bc1337b201ff87d
I also added a link to this section from `Report::new`: 7e5d08c9d55b3729e7f61e17c2ce69ac55fe330c